src/app.py

```diff
@@ -795,14 +818,26 @@ def publish_datastage(identifier):
                 return Response(f"{dataset_uuid}: no donor found for dataset, will not Publish")
 
             #get info for the dataset to be published
-            q = f"MATCH (e:Dataset {{uuid: '{dataset_uuid}'}}) RETURN e.uuid as uuid, e.entity_type as entitytype, e.status as status, e.data_access_level as data_access_level, e.group_uuid as group_uuid, e.contacts as contacts, e.contributors as contributors"
+            q = f"MATCH (e:Dataset {{uuid: '{dataset_uuid}'}}) RETURN " \
+                "e.uuid as uuid, e.entity_type as entitytype, e.status as status, " \
+                "e.data_access_level as data_access_level, e.group_uuid as group_uuid, " \
+                "e.contacts as contacts, e.contributors as contributors"
+            if is_primary:
+                q += ", e.ingest_metadata as ingest_metadata"
             rval = neo_session.run(q).data()
             dataset_status = rval[0]['status']
             dataset_entitytype = rval[0]['entitytype']
             dataset_data_access_level = rval[0]['data_access_level']
             dataset_group_uuid = rval[0]['group_uuid']
             dataset_contacts = rval[0]['contacts']
             dataset_contributors = rval[0]['contributors']
+            dataset_ingest_matadata_dict = None
+            if is_primary:
+                dataset_ingest_metadata = rval[0].get('ingest_metadata')
+                if dataset_ingest_metadata is not None:
+                    dataset_ingest_matadata_dict: dict =\
+                        string_helper.convert_str_literal(dataset_ingest_metadata)
```

Comment:
Since the commons code raises the exceptions (https://github.com/hubmapconsortium/commons/pull/116/files#diff-335d97171e5c7b7e0fe029af45da8fa821532151282425f62fc3fea0361974d8R101-R103)
```
except (SyntaxError, ValueError, TypeError) as e:
    raise ValueError(f"Invalid expression (string value): {data_str} from ast.literal_eval(); "
        f"specific error: {str(e)}")
```
I would expect the exceptions to be caught here and handled accordingly.

Reply:
@yuanzhou In general I don't think we should need to do this for non-specific exceptions.  There should be generic handling to log and send back a 500 error on unexpected exceptions.